plutus-core/plutus-core/src/PlutusCore/Name/UniqueSet.hs

```diff
@@ -0,0 +1,91 @@
+module PlutusCore.Name.UniqueSet (
+  UniqueSet (..),
+  insertByUnique,
+  insertByName,
+  singletonName,
+  fromFoldable,
+  fromUniques,
+  fromNames,
+  memberByUnique,
+  memberByName,
+  notMemberByName,
+  (\\),
+  union,
+  setOfByUnique,
+  setOfByName,
+) where
+
+import Control.Lens (Getting, view)
+import Control.Lens.Getter (views)
+import Data.Coerce (Coercible, coerce)
+import Data.Foldable (foldl')
+import Data.IntSet qualified as IS
+import Data.IntSet.Lens qualified as IS
+import PlutusCore.Name.Unique (HasUnique (..), Unique (Unique))
+
+-- | A set containing uniques.
+newtype UniqueSet unique = UniqueSet
+  { unUniqueSet :: IS.IntSet
+  }
+  deriving newtype (Show, Eq, Semigroup, Monoid)
+
+-- | Insert a value by a unique.
```

Comment:
These comments read as more "map-y" (indeed, I think they're just copied from the map version). I think it could just be "insert a unique". And similarly for the other comments.